framework/src/main/java/com/github/dbmdz/flusswerk/framework/exceptions/SkipProcessingException.java

```diff
@@ -0,0 +1,71 @@
+package com.github.dbmdz.flusswerk.framework.exceptions;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+
+import com.github.dbmdz.flusswerk.framework.model.Message;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/** Indicates that processing should be skipped from the point on where this exception is thrown. */
+public class SkipProcessingException extends RuntimeException {
+
+  private final List<Message> outgoingMessages = new ArrayList<>();
+
+  /**
+   * Skip processing with an explanation and a cause.
+   *
+   * @param message The explanation why skipping makes sense here.
+   * @param cause The cause why skipping should happen.
+   */
+  public SkipProcessingException(String message, Throwable cause) {
```

Comment:
Is skipping because of an Exception something that should be encouraged via the API?

Reply:
Good point, this would actually go against the Flusswerk philosophy - either processing is ok (then there's  no exception needed), or we have to deal with it explicitely (with `StopProcessingException` or `RetryProcessingException`). Try and ignore is a very unusual usecase that should be handled by the app itself.

Reply:
If this would be really needed by an app there is always `exception.initCause(e)`.